src/bp/core/services/converse.ts

```diff
@@ -170,13 +165,17 @@ export class ConverseService {
     })
   }
 
-  private _handleCapturePayload(event: IO.Event) {
+  private async _handleCapturePayload(event: IO.OutgoingEvent) {
     const userKey = buildUserKey(event.botId, event.target)
     if (!this._responseMap[userKey]) {
       this._responseMap[userKey] = { responses: [] }
     }
 
     this._responseMap[userKey].responses!.push(event.payload)
+
+    if (event.type !== 'typing' && event.type !== 'data') {
+      await this.messagingApi.createMessage(+event.threadId!, event.id, event.incomingEventId!, 'bot', event.payload)
```

Comment:
is it guaranteed that threadId is a parsable number ?

Reply:
It's guaranteed if the converse api is used in a normal way. It always gets or create a conversation before sending a message and the `threadId` is just de id of the conversation.

The only way to make this crash would be to send a synthetic event through the even engine for the converse channel and having the threadId be some made up id.